tasks/components/layout/TasksKanbanBoard.tsx

```diff
@@ -122,26 +122,29 @@ export const TasksKanbanBoard = ({
     const activeItems = sortedTasks[activeColumn]
     const overItems = sortedTasks[overColumn]
 
-    // Find the indexes for the items
+    // Item index (order) of the column, which the user drags the task from
     const activeIndex = activeItems.findIndex(item => item.id === active.id)
+    // Item index (order) of the column, which the user drags the task to
     const overIndex = overItems.findIndex(item => item.id !== over?.id)
 
-    if (activeIndex === -1 || overIndex === -1) {
+    // If it can't identify the dragged item
+    if (activeIndex === -1) {
       return
     }
 
+    // -1 means, that there was no task found, which you dragged your task over
+    const insertAtIndex = overIndex === -1 ? 0 : overIndex
+    // Insert task at insertAtIndex in the overColumn
+    sortedTasks[overColumn].splice(insertAtIndex, 0, sortedTasks[activeColumn][activeIndex] as TaskDTO)
```

Comment:
don't mutate the array here (i.e. use `slice` instead and save a copy of the array)

Reply:
@JannikWibker, @konstantinschuette and myself are not quite sure about the benefit. Could you propose a change?

Reply:
[State in react is immutable](https://react.dev/learn/updating-objects-in-state). Doing modifications anyways leads to undefined behaviour / is not supported at all.

```tsx
const newOverColumn = sortedTasks[overColumn].slice(insertAtIndex, 0, sortedTasks[activeColumn][activeIndex] as TaskDTO)

// The last array access is safe as we checked the index before (> -1 and < length; findIndex will not return something >= length)
// Remove dragged task from its origin column, which is the activeColumn
const newActiveColumn = sortedTasks[activeColumn]
  .map((task, i) => i === activeIndex ? { ...task, status: overColumn } : task)
  .filter((item) => item.id !== active.id)

setSortedTasks({
  ...sortedTasks,
  [overColumn]; newOverColumn,
  [activeColumn]: newActiveColumn
})
```

Reply:
Status? Current tasks version is not usable though...